onnxruntime/core/optimizer/qdq_transformer/qdq_propagation.cc

```diff
@@ -20,54 +23,108 @@ bool CanNodePropagate(const Node& node) {
          graph_utils::IsSupportedOptypeVersionAndDomain(node, "Reshape", {5, 13, 14, 19}) ||
          graph_utils::IsSupportedOptypeVersionAndDomain(node, "Transpose", {1, 13}) ||
          graph_utils::IsSupportedOptypeVersionAndDomain(node, "Squeeze", {1, 11, 13}) ||
-         graph_utils::IsSupportedOptypeVersionAndDomain(node, "Unsqueeze", {1, 11, 13});
+         graph_utils::IsSupportedOptypeVersionAndDomain(node, "Unsqueeze", {1, 11, 13}) ||
+         graph_utils::IsSupportedOptypeVersionAndDomain(node, "Slice", {1, 10, 11, 13});
 }
 
-// convert this: src_node -> dst_node
-// to this:      src_node -> Q -> DQ -> dst_node
-// assumptions:
-// 1. insertion_edge is valid - node indexes refer to valid nodes, arg name refers to a valid NodeArg, and it
-//    corresponds to an actual graph relationship
-// 2. scale_initializer_nodearg and zp_initializer_nodearg_ptr (if not null) are constant initializers
-Status InsertQDQPair(Graph& graph, const ExtendedGraphEdge& insertion_edge,
-                     NodeArg& scale_initializer_nodearg, NodeArg* zp_initializer_nodearg_ptr,
-                     const std::string& qdq_domain, const logging::Logger& logger) {
-  auto* src_node = insertion_edge.GetMutableNodeAtEnd(graph, ExtendedGraphEdge::End::Source);
-  auto* dst_node = insertion_edge.GetMutableNodeAtEnd(graph, ExtendedGraphEdge::End::Destination);
-
-  ORT_ENFORCE(src_node || dst_node, "At least one graph node must be specified in the propagation edge.");
-
-  const auto& base_name = insertion_edge.arg_name;
-  auto& base_node_arg = *graph.GetNodeArg(base_name);
+// Validates edges into which to insert Q -> DQ ops.
+// - Must have at least one edge.
+// - All edges with a source node must originate from the same source node's output.
+// - All edges must be attached to either a source node or a destination node.
+Status ValidateQDQInsertionEdges(Graph& graph, const InlinedVector<ExtendedGraphEdge>& insertion_edges) {
```

Comment:
use `span<const T>` where applicable, e.g., in place of `const Container<T>&`
```suggestion
Status ValidateQDQInsertionEdges(Graph& graph, gsl::span<const ExtendedGraphEdge> insertion_edges) {
```

Reply:
Fixed